### PR TITLE
[All hosts](security) Update NAA article to support a walkthrough

### PR DESCRIPTION
**Pull request by davidchesnut**
Reworked the code and steps so that you could follow this as a walkthrough and paste code into a yo office project and see it work.

### CODE REVIEW

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md [hunk 1]
```diff
@@ -18,6 +17,12 @@ You can use the MSAL.js library (version 3.11 and later) with nested app authent
 
 > [!IMPORTANT]
 > Nested app authentication (NAA) is currently in preview. To try this feature, join the Microsoft 365 Insider Program (https://insider.microsoft365.com/join) and choose the Beta Channel. Don't use NAA in production add-ins. We invite you to try out NAA in test or development environments and welcome feedback on your experience through GitHub (see the **Feedback** section at the end of this page).
+> NAA is supported in the following builds.
+>
+> - Word, Excel, and PowerPoint on Windows build 16.0.17531.20000 or later.
+> - Word, Excel, and PowerPoint on Mac build 16.85.24040319 or later.
+> - Outlook on Windows build 16.0.17531.20000 or later.
+> - Outlook on Mac build 16.85.24040319 or later.
```
Comment (Rick-Kirkham) [inline docs/develop/enable-nested-app-authentication-in-your-add-in.md:25]: Why list Outlook separately? The build numbers are the same as for WXP.

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md [hunk 2]
```diff
@@ -41,30 +46,52 @@ Trusted broker groups are dynamic by design and can be updated in the future to
...
-Office.onReady(async (info) => {
-    pca = await msalBrowser.PublicClientNext.createPublicClientApplication(msalConfig);
+  ```javascript
+  Office.onReady(async (info) => {
+    if (info.host === Office.HostType.Excel) {
+      document.getElementById("sideload-msg").style.display = "none";
+      document.getElementById("app-body").style.display = "flex";
+      document.getElementById("run").onclick = run;
+
+      // Initialize the publice client application
```
Comment (Rick-Kirkham) [inline docs/develop/enable-nested-app-authentication-in-your-add-in.md:90]:
```suggestion
      // Initialize the public client application
```

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md [hunk 3]
```diff
@@ -76,114 +103,159 @@ The following steps show the pattern to use for acquiring a token.
...
 
 ### Have a fallback when NAA isn't supported
 
-While we strive to provide a high-degree of compatibility with these flows across the Microsoft ecosystem, your application may appear in downlevel/legacy clients that haven't been updated to support NAA. In these cases, your application won't support seamless SSO and you may need to invoke special APIs for interacting with the user to open authentication dialogs. For more information, see [Authenticate and authorize with the Office dialog API](/office/dev/add-ins/develop/auth-with-office-dialog-api).
+While we strive to provide a high-degree of compatibility with these flows across the Microsoft ecosystem, your add-in may be loaded in an older Office host that does not support NAA. In these cases, your add-in won't support seamless SSO and you may need to fall back to an alternate method of authenticating the user. In generaly you'll want to use the MSAL SPA authentication pattern with the Office JS dialog API. For more information, see the following resources.
```
Comment (Rick-Kirkham) [inline docs/develop/enable-nested-app-authentication-in-your-add-in.md:218]:
```suggestion
While we strive to provide a high-degree of compatibility with these flows across the Microsoft ecosystem, your add-in may be loaded in an older Office host that does not support NAA. In these cases, your add-in won't support seamless SSO and you may need to fall back to an alternate method of authenticating the user. In general you'll want to use the MSAL SPA authentication pattern with the [Office JS dialog API](auth-with-office-dialog-api.md). For more information, see the following resources.
```

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]
I haven't actually gone through the steps yet. I'll try to today or tomorrow.

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md (hunk 1 above)
Comment (lindalu-MSFT) [reply docs/develop/enable-nested-app-authentication-in-your-add-in.md:25]: Please remove period on bulleted items. You only use when items are sentences.

**Review by lindalu-MSFT — APPROVED** [review]

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md (hunk 1 above)
Comment (davidchesnut) [reply docs/develop/enable-nested-app-authentication-in-your-add-in.md:25]: fixed, thanks!

**Review by Rick-Kirkham — CHANGES_REQUESTED** [review]
A few comments. One big point: having a walkthrough is a good idea, but I think it should be a separate article from the overview. If you keep it in this article, I recommend that you move it to the end of the article. As the article is structured now, the overview information comes after the walkthrough, which I think is confusing and some readers who are just looking for overview info, might think that there isn't any more information beyond the small bit that precedes the walkthrough.

File: docs/develop/enable-nested-app-authentication-in-your-add-in.md [hunk 4]
```diff
@@ -41,30 +46,52 @@ Trusted broker groups are dynamic by design and can be updated in the future to
 
 Configure your add-in to use NAA by setting the `supportsNestedAppAuth` property to true in your MSAL configuration. This enables MSAL to use APIs on its native application host (for example, Outlook) to acquire tokens for your application. If you don't set this property, MSAL uses the default JavaScript-based implementation to acquire tokens for your application, which may lead to unexpected auth prompts and unsatisfiable conditional access policies when running inside of a webview.
 
-```JavaScript
-// Configuration for NAA.  
+The following steps show how to enable NAA in the `taskpane.js` or `taskpane.ts` file in a project built with `yo office`. The code can be adapted to any project.
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply docs/develop/enable-nested-app-authentication-in-your-add-in.md:47]: I'll work to reword that better.

**Review by Rick-Kirkham — APPROVED** [review]
Approving, although I still recommend splitting off the walkthrough to a different article.

**Pull request merged**